pkg/build/builder/sti.go

```diff
@@ -119,10 +169,25 @@ func (s *STIBuilder) Build() error {
 		)
 		if authPresent {
 			glog.Infof("Using provided push secret for pushing %s image", tag)
+		} else {
+			glog.Infof("No push secret provided")
 		}
 		glog.Infof("Pushing %s image ...", tag)
 		if err := pushImage(s.dockerClient, tag, pushAuthConfig); err != nil {
-			return fmt.Errorf("Failed to push image: %v", err)
+			// write extended error message to assist in problem resolution
+			msg := fmt.Sprintf("Failed to push image. Response from registry is: %v", err)
+			if authPresent {
+				glog.Infof("Registry server Address: %s", pushAuthConfig.ServerAddress)
```

Comment:
@mfojtik I though you did something similar previously, why it was removed, do you recall something?